### PR TITLE
*: rename statement summary tables

### PR DESCRIPTION
**Pull request by djshow832**
<!-- Thank you for contributing to TiDB!

PR Title Format:
1. pkg [, pkg2, pkg3]: what's changed
2. *: what's changed

-->

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/15348

Problem Summary:
Statement summary tables are different with that in MySQL, and we can't make them the same, so renaming it should be better.
Because tables in `performance_schema` should be picoseconds, but our tables, including slow_query, all use nanoseconds. So it's better to move them into `information_schema`.

### What is changed and how it works?

What's Changed:
Move all statement summary tables into `information_schema` and name them like `statements_summary`.
Leave `events_statements_summary_by_digest` empty.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List <!--REMOVE the items that are not applicable-->

Tests <!-- At least one of them must be included. -->

- Unit test
- Manual test (add detailed scripts or steps below)
```
mysql> show create table information_schema.statements_summary;
+--------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table              | Create Table                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
+--------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| STATEMENTS_SUMMARY | CREATE TABLE `STATEMENTS_SUMMARY` (
  `SUMMARY_BEGIN_TIME` timestamp NOT NULL COMMENT 'Begin time of this summary',
  `SUMMARY_END_TIME` timestamp NOT NULL COMMENT 'End time of this summary',
  `STMT_TYPE` varchar(64) NOT NULL COMMENT 'Statement type',
  `SCHEMA_NAME` varchar(64) NOT NULL COMMENT 'Current schema',
  `DIGEST` varchar(64) NOT NULL,
  `DIGEST_TEXT` text NOT NULL COMMENT 'Normalized statement',
  `TABLE_NAMES` text DEFAULT NULL COMMENT 'Involved tables',
  `INDEX_NAMES` text DEFAULT NULL COMMENT 'Used indices',
  `SAMPLE_USER` varchar(64) DEFAULT NULL COMMENT 'Sampled user who executed these statements',
  `EXEC_COUNT` bigint(20) unsigned NOT NULL COMMENT 'Count of executions',
  `SUM_LATENCY` bigint(20) unsigned NOT NULL COMMENT 'Sum latency of these statements',
  `MAX_LATENCY` bigint(20) unsigned NOT NULL COMMENT 'Max latency of these statements',
  `MIN_LATENCY` bigint(20) unsigned NOT NULL COMMENT 'Min latency of these statements',
  `AVG_LATENCY` bigint(20) unsigned NOT NULL COMMENT 'Average latency of these statements',
  `AVG_PARSE_LATENCY` bigint(20) unsigned NOT NULL COMMENT 'Average latency of parsing',
  `MAX_PARSE_LATENCY` bigint(20) unsigned NOT NULL COMMENT 'Max latency of parsing',
  `AVG_COMPILE_LATENCY` bigint(20) unsigned NOT NULL COMMENT 'Average latency of compiling',
  `MAX_COMPILE_LATENCY` bigint(20) unsigned NOT NULL COMMENT 'Max latency of compiling',
  `COP_TASK_NUM` bigint(20) unsigned NOT NULL COMMENT 'Total number of CopTasks',
  `AVG_COP_PROCESS_TIME` bigint(20) unsigned NOT NULL COMMENT 'Average processing time of CopTasks',
  `MAX_COP_PROCESS_TIME` bigint(20) unsigned NOT NULL COMMENT 'Max processing time of CopTasks',
  `MAX_COP_PROCESS_ADDRESS` varchar(256) DEFAULT NULL COMMENT 'Address of the CopTask with max processing time',
  `AVG_COP_WAIT_TIME` bigint(20) unsigned NOT NULL COMMENT 'Average waiting time of CopTasks',
  `MAX_COP_WAIT_TIME` bigint(20) unsigned NOT NULL COMMENT 'Max waiting time of CopTasks',
  `MAX_COP_WAIT_ADDRESS` varchar(256) DEFAULT NULL COMMENT 'Address of the CopTask with max waiting time',
  `AVG_PROCESS_TIME` bigint(20) unsigned NOT NULL COMMENT 'Average processing time in TiKV',
  `MAX_PROCESS_TIME` bigint(20) unsigned NOT NULL COMMENT 'Max processing time in TiKV',
  `AVG_WAIT_TIME` bigint(20) unsigned NOT NULL COMMENT 'Average waiting time in TiKV',
  `MAX_WAIT_TIME` bigint(20) unsigned NOT NULL COMMENT 'Max waiting time in TiKV',
  `AVG_BACKOFF_TIME` bigint(20) unsigned NOT NULL COMMENT 'Average waiting time before retry',
  `MAX_BACKOFF_TIME` bigint(20) unsigned NOT NULL COMMENT 'Max waiting time before retry',
  `AVG_TOTAL_KEYS` bigint(20) unsigned NOT NULL COMMENT 'Average number of scanned keys',
  `MAX_TOTAL_KEYS` bigint(20) unsigned NOT NULL COMMENT 'Max number of scanned keys',
  `AVG_PROCESSED_KEYS` bigint(20) unsigned NOT NULL COMMENT 'Average number of processed keys',
  `MAX_PROCESSED_KEYS` bigint(20) unsigned NOT NULL COMMENT 'Max number of processed keys',
  `AVG_PREWRITE_TIME` bigint(20) unsigned NOT NULL COMMENT 'Average time of prewrite phase',
  `MAX_PREWRITE_TIME` bigint(20) unsigned NOT NULL COMMENT 'Max time of prewrite phase',
  `AVG_COMMIT_TIME` bigint(20) unsigned NOT NULL COMMENT 'Average time of commit phase',
  `MAX_COMMIT_TIME` bigint(20) unsigned NOT NULL COMMENT 'Max time of commit phase',
  `AVG_GET_COMMIT_TS_TIME` bigint(20) unsigned NOT NULL COMMENT 'Average time of getting commit_ts',
  `MAX_GET_COMMIT_TS_TIME` bigint(20) unsigned NOT NULL COMMENT 'Max time of getting commit_ts',
  `AVG_COMMIT_BACKOFF_TIME` bigint(20) unsigned NOT NULL COMMENT 'Average time before retry during commit phase',
  `MAX_COMMIT_BACKOFF_TIME` bigint(20) unsigned NOT NULL COMMENT 'Max time before retry during commit phase',
  `AVG_RESOLVE_LOCK_TIME` bigint(20) unsigned NOT NULL COMMENT 'Average time for resolving locks',
  `MAX_RESOLVE_LOCK_TIME` bigint(20) unsigned NOT NULL COMMENT 'Max time for resolving locks',
  `AVG_LOCAL_LATCH_WAIT_TIME` bigint(20) unsigned NOT NULL COMMENT 'Average waiting time of local transaction',
  `MAX_LOCAL_LATCH_WAIT_TIME` bigint(20) unsigned NOT NULL COMMENT 'Max waiting time of local transaction',
  `AVG_WRITE_KEYS` double unsigned NOT NULL COMMENT 'Average count of written keys',
  `MAX_WRITE_KEYS` bigint(20) unsigned NOT NULL COMMENT 'Max count of written keys',
  `AVG_WRITE_SIZE` double unsigned NOT NULL COMMENT 'Average amount of written bytes',
  `MAX_WRITE_SIZE` bigint(20) unsigned NOT NULL COMMENT 'Max amount of written bytes',
  `AVG_PREWRITE_REGIONS` double unsigned NOT NULL COMMENT 'Average number of involved regions in prewrite phase',
  `MAX_PREWRITE_REGIONS` int(11) unsigned NOT NULL COMMENT 'Max number of involved regions in prewrite phase',
  `AVG_TXN_RETRY` double unsigned NOT NULL COMMENT 'Average number of transaction retries',
  `MAX_TXN_RETRY` int(11) unsigned NOT NULL COMMENT 'Max number of transaction retries',
  `SUM_BACKOFF_TIMES` bigint(20) unsigned NOT NULL COMMENT 'Sum of retries',
  `BACKOFF_TYPES` varchar(1024) DEFAULT NULL COMMENT 'Types of errors and the number of retries for each type',
  `AVG_MEM` bigint(20) unsigned NOT NULL COMMENT 'Average memory(byte) used',
  `MAX_MEM` bigint(20) unsigned NOT NULL COMMENT 'Max memory(byte) used',
  `AVG_AFFECTED_ROWS` double unsigned NOT NULL COMMENT 'Average number of rows affected',
  `FIRST_SEEN` timestamp NOT NULL COMMENT 'The time these statements are seen for the first time',
  `LAST_SEEN` timestamp NOT NULL COMMENT 'The time these statements are seen for the last time',
  `QUERY_SAMPLE_TEXT` text DEFAULT NULL COMMENT 'Sampled original statement',
  `PREV_SAMPLE_TEXT` text DEFAULT NULL COMMENT 'The previous statement before commit',
  `PLAN_DIGEST` varchar(64) DEFAULT NULL COMMENT 'Digest of its execution plan',
  `PLAN` text DEFAULT NULL COMMENT 'Sampled execution plan'
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+--------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)

mysql> select * from information_schema.statements_summary limit 1\G
*************************** 1. row ***************************
       SUMMARY_BEGIN_TIME: 2020-04-08 14:30:00
         SUMMARY_END_TIME: 2020-04-08 15:00:00
                STMT_TYPE: show
              SCHEMA_NAME: test
                   DIGEST: 2aa3a18dc0da0e3ec7aef05f7dec9a2887574a5130a937c1899e1458ab8ced8f
              DIGEST_TEXT: show create table information_schema . statements_summary
              TABLE_NAMES: information_schema.statements_summary
              INDEX_NAMES: NULL
              SAMPLE_USER: root
               EXEC_COUNT: 1
              SUM_LATENCY: 2630462
              MAX_LATENCY: 2630462
              MIN_LATENCY: 2630462
              AVG_LATENCY: 2630462
        AVG_PARSE_LATENCY: 42360
        MAX_PARSE_LATENCY: 42360
      AVG_COMPILE_LATENCY: 1254596
      MAX_COMPILE_LATENCY: 1254596
             COP_TASK_NUM: 0
     AVG_COP_PROCESS_TIME: 0
     MAX_COP_PROCESS_TIME: 0
  MAX_COP_PROCESS_ADDRESS: NULL
        AVG_COP_WAIT_TIME: 0
        MAX_COP_WAIT_TIME: 0
     MAX_COP_WAIT_ADDRESS: NULL
         AVG_PROCESS_TIME: 0
         MAX_PROCESS_TIME: 0
            AVG_WAIT_TIME: 0
            MAX_WAIT_TIME: 0
         AVG_BACKOFF_TIME: 0
         MAX_BACKOFF_TIME: 0
           AVG_TOTAL_KEYS: 0
           MAX_TOTAL_KEYS: 0
       AVG_PROCESSED_KEYS: 0
       MAX_PROCESSED_KEYS: 0
        AVG_PREWRITE_TIME: 0
        MAX_PREWRITE_TIME: 0
          AVG_COMMIT_TIME: 0
          MAX_COMMIT_TIME: 0
   AVG_GET_COMMIT_TS_TIME: 0
   MAX_GET_COMMIT_TS_TIME: 0
  AVG_COMMIT_BACKOFF_TIME: 0
  MAX_COMMIT_BACKOFF_TIME: 0
    AVG_RESOLVE_LOCK_TIME: 0
    MAX_RESOLVE_LOCK_TIME: 0
AVG_LOCAL_LATCH_WAIT_TIME: 0
MAX_LOCAL_LATCH_WAIT_TIME: 0
           AVG_WRITE_KEYS: 0
           MAX_WRITE_KEYS: 0
           AVG_WRITE_SIZE: 0
           MAX_WRITE_SIZE: 0
     AVG_PREWRITE_REGIONS: 0
     MAX_PREWRITE_REGIONS: 0
            AVG_TXN_RETRY: 0
            MAX_TXN_RETRY: 0
        SUM_BACKOFF_TIMES: 0
            BACKOFF_TYPES: NULL
                  AVG_MEM: 0
                  MAX_MEM: 0
        AVG_AFFECTED_ROWS: 0
               FIRST_SEEN: 2020-04-08 14:48:38
                LAST_SEEN: 2020-04-08 14:48:38
        QUERY_SAMPLE_TEXT: show create table information_schema.statements_summary
         PREV_SAMPLE_TEXT:
              PLAN_DIGEST: a5476f749f622419e9f82d16f5f927ba132e6d509388f8f14d3048af8226fcd4
                     PLAN: 	Show_2	root	1
1 row in set (0.00 sec)
```

Side effects

- Breaking backward compatibility

### Release note <!-- bugfixes or new feature need a release note -->

 - Move statement summary tables into `information_schema` and rename them to `xxx_statements_summary_xxx`.

### CODE REVIEW

File: infoschema/perfschema/tables.go [hunk 1]
```diff
@@ -202,19 +189,7 @@ func (vt *perfSchemaTable) Type() table.Type {
...
-	if pm := privilege.GetPrivilegeManager(ctx); pm != nil {
-		isSuper = pm.RequestVerificationWithUser("", "", "", mysql.SuperPriv, user)
-	}
-
 	switch vt.meta.Name.O {
-	case tableNameEventsStatementsSummaryByDigest:
-		fullRows = stmtsummary.StmtSummaryByDigestMap.ToCurrentDatum(user, isSuper)
-	case tableNameEventsStatementsSummaryByDigestHistory:
-		fullRows = stmtsummary.StmtSummaryByDigestMap.ToHistoryDatum(user, isSuper)
 	case tableNameTiDBProfileCPU:
```
Comment (Deardrops) [inline infoschema/perfschema/tables.go:193]: Do we need to check privilege before use access this table?

Comment (djshow832) [reply infoschema/perfschema/tables.go:193]: Privilege management on statement summary tables is very special. All users can see statements executed by themselves, and superusers can see all statements. So we can only check privilege in `ToCurrentDatum` and `ToHistoryDatum`.

Comment (djshow832) [reply infoschema/perfschema/tables.go:193]: This table has nothing to do with this PR.

File: docs/design/2019-11-14-tidb-builtin-diagnostics-zh_CN.md [hunk 2]
```diff
@@ -289,7 +289,7 @@ message ServerInfoResponse {
 
 - 终端用户：用户直接通过 SQL 查询获取集群信息排查问题
 - 运维系统：TiDB 的使用环境比较多样，客户可以通过 SQL 获取集群信息将 TiDB 集成到自己的运维系统中
-- 生态工具：外部工具通过 SQL 拿到集群信息实现功能定制，比如 `[sqltop](https://github.com/ngaut/sqltop)` 可以直接通过集群 `events_statements_summary_by_digest` 获取整个集群的 SQL 采样信息
+- 生态工具：外部工具通过 SQL 拿到集群信息实现功能定制，比如 `[sqltop](https://github.com/ngaut/sqltop)` 可以直接通过集群 `statements_summary` 获取整个集群的 SQL 采样信息
```
Comment (breeswish) [inline docs/design/2019-11-14-tidb-builtin-diagnostics-zh_CN.md:292]: this no longer stands

Comment (djshow832) [reply docs/design/2019-11-14-tidb-builtin-diagnostics-zh_CN.md:292]: Why? It seems this tool can access new `statements_summary` table. Do you mean we can access `cluster_statements_summary` directly?

**Pull request merged**